npm/webpack-dev-server-fresh/src/helpers/nextHandler.ts

```diff
@@ -0,0 +1,146 @@
+import type { CreateFinalWebpackConfig } from '../createWebpackDevServer'
+import debugLib from 'debug'
+import type { Configuration } from 'webpack'
+import * as fs from 'fs'
+import * as path from 'path'
+
+type PresetHandler = Omit<CreateFinalWebpackConfig, 'frameworkConfig'>
+
+const debug = debugLib('cypress:webpack-dev-server-fresh:nextHandler')
+
+export async function nextHandler ({ devServerConfig, sourceWebpackModulesResult }: PresetHandler) {
+  const webpackConfig = await loadWebpackConfig({ devServerConfig, sourceWebpackModulesResult })
```

Comment:
This look really thorough, nice job -- it's a little challenging to fully understand what case each branch handles, might be a good candidate for some fairly verbose comments?

Reply:
Added some more comments throughout